Automattic/woocommerce-payments
# Expose charge id so merchant can use it to submit duplicate dispute evidence

Fixes https://github.com/Automattic/woocommerce-payments/issues/5141.

Currently, when merchant gets a dispute with reason 'duplicate', a field to specify a duplicate charge id will appear in the evidence form. However, per our recent work where we removed dispute details page and embed dispute information into payment details page, charge id of a transaction is not shown anywhere.

> [!note]
> TODOs:
> - A docs page needs to be created to explain about duplicate charge info section.
> - We need to update the docs link in this PR diff to point to that newly created docs page.

#### Changes proposed in this Pull Request

On transaction details page:
* Show charge id.

![Screenshot 2024-02-01 at 01 19 24](https://github.com/Automattic/woocommerce-payments/assets/73803630/aad0f36e-80d1-451d-ae3a-994561a72de1)


On dispute evidence form:
* Show documentation link (will be provided later) to explain more about duplicate charge info.
* Show text to inform merchant that charge id starts with `ch_`. This is to satisfy this suggestion https://github.com/Automattic/woocommerce-payments/issues/5141#issuecomment-1317725678.

Before:
![Screenshot 2024-02-01 at 01 18 11](https://github.com/Automattic/woocommerce-payments/assets/73803630/d4216ec6-9342-4ce3-84f9-c6cc17df80c9)

After:
![Screenshot 2024-02-01 at 01 23 04](https://github.com/Automattic/woocommerce-payments/assets/73803630/cc4080c5-08cd-417a-a580-eb1a5d7fb0e0)

#### Testing instructions

* Purchase a product with a card that will trigger a dispute, eg `4000000000000259`.
* Because there is no testing card that will trigger a dispute with reason `duplicate`, we need to modify the reason in the code for testing purposes. After this [line](https://github.com/Automattic/woocommerce-payments/blob/bbf6e1f00d55f44e40a32eacfe38b19be5bca60e/includes/wc-payment-api/class-wc-payments-api-client.php#L516), add:
```php
$dispute['reason'] = 'duplicate';
```
* Go to Payments > Disputes > click on the new dispute.
* You will be directed to the payment detail page. Confirm that Charge ID now shows up on the top right.
* Click on the 'Challenge dispute' button.
* You will be directed to the evidence form page. Choose Product Type 'Multiple product types'.
* Confirm that under the 'Duplicate charge info' section, there is a link to a docs and that there is a text to indicate that charge id starts with `ch_`.

-------------------


**Post merge**
